### ISSUE TITLE
Broken basic authentication with libcurl=7.85

### ISSUE THREAD

**Issue by jonashaag** [mamba-org/mamba#2010]
We have the following in `.condarc`:

```
channel_alias: https:/some%40emailaddress.com:password@somehost...
```

With Micromamba/Mamba 0.27 (= curl>=7.85) this doesn't work because the `Authorization` header sent contains `%40` while it should be `@`.

It seems to be a problem with curl 7.84 or 7.85 because Mamba with curl 7.83 it works. We don't have a release with 0.74 so I can't easily test.

Is this a regression in curl? I couldn't find anything in the curl release notes.

**Comment by wolfv** [comment]
hmm, did you try a literal `@`? Not sure what's going on exactly.

**Comment by wolfv** [comment]
Also, I thought we have a test, at least for `mamba` that covers this :/

**Comment by jonashaag** [comment]
I can reproduce this with `reposerver.py` with Micromamba `main`. curl 7.83 works, curl 7.85 does not. It only happens when reading from `.condarc`, not when passing via `-c` on the command line.

**Comment by jonashaag** [comment]
@AdrianFreundQC I'm working on a fix for this in #2012. Can you please extend the proxy tests to cover this case as well? I quickly tried the proxy tests with `%40` in the username and it seemed to work fine.

**Comment by wolfv** [comment]
I just added a quick test to `test_url.cpp` and this one seemed fine:

```
        std::string input3 = "https://some%40emailaddress.com:password@somehost.com/channel1";

        split_scheme_auth_token(input3, remaining_url, scheme, auth, token);
        EXPECT_EQ(remaining_url, "somehost.com/channel1");
        EXPECT_EQ(scheme, "https");
        EXPECT_EQ(auth, "some%40emailaddress.com:password");
        EXPECT_EQ(token, "");
```

Not sure yet where it fails.

**Comment by jonashaag** [comment]
Curl issue that will be fixed with the next release.

We should make a Micromamba release as soon as the next curl release is available – or maybe is it worth using a patched version of curl 7.85?
